### ISSUE TITLE
Select from namespace doesn't issue error

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#14689]
Consider the following code:

``` cs
using System.Linq;

namespace ConsoleApp
{
    class Program
    {
        static void Main(string[] args)
        {
        }

        public static int Test()
        {
            var x = from c in ConsoleApp
                    select 3;
            return x.First();
        }
    }
}
```

This code is invalid and fails to compile.  The error though is the emit catch all "failed to emit".  That means the compiler is issuing an error earlier and not surfacing it.  In larger programs that makes the problem very difficult to track down.  Need to correct this.

**Comment by jaredpar** [comment]
Note that it's possible this error has already been fixed.  I tried the repro on VS2015U3 but not on Dev15 yet.

**Comment by OmarTawfik** [comment]
Could repro in Dev15 Preview 5. Will look into a fix now.
